docs/development/setup.md

```diff
@@ -0,0 +1,143 @@
+# Local Development Setup
+
+- [Prerequisites](#prerequisites)
+- [Preperation](#preperation)
+- [Run libvirt-provider for local development](#run-libvirt-provider-for-local-development)
+- [Interact with the `libvirt-provider`](#interact-with-the-libvirt-provider)
+- [Deploy `libvirt-provider`](#deploy-libvirt-provider)
+
+> ℹ️ **NOTE**:</br>
+> To be able to take exec console of the machine, you can follow any one of the below approaches:</br>
+> - Run the `libvirt-provider` as the `libvirt-qemu` user.</br>
+> - Create an entry with `devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=0660 0 0` in `/etc/fstab`.</br>
+> - Manually ensure that you have `0660` access permissions on the character files created in `/dev/pts`.</br>
+
+## Prerequisites
+
+- Linux (code contains OS specific code)
+- go >= 1.20
+- `git`, `make` and `kubectl`
+- Access to a Kubernetes cluster ([Minikube](https://minikube.sigs.k8s.io/docs/), [kind](https://kind.sigs.k8s.io/) or a
+  real cluster)
+- [libvirt](http://libvirt.org)
+- [QEMU](https://www.qemu.org/download/)
+- `irictl-machine` should be running locally or as [container](https://github.com/ironcore-dev/ironcore/pkgs/container/ironcore-irictl-machine)
+
+## Preperation
+
+### Setup `irictl-machine`
+
+1. **Clone ironcore repository**
+
+    ```bash
+    git clone git@github.com:ironcore-dev/ironcore.git
+    cd ironcore
+    ```
+
+1. **Build `irictl-machine`**
+
+    ```bash
+    go build -o bin/irictl-machine ./irictl-machine/cmd/irictl-machine/main.go
+    ```
+
+## Run libvirt-provider for local development
+
+1. **Clone the Repository**
+
+    To bring up and start locally the libvirt-provider project for development purposes you first need to clone the repository.
+
+    ```bash
+    git clone git@github.com:ironcore-dev/libvirt-provider.git
+    cd libvirt-provider
+    ```
+
+1. **Build the `libvirt-provider`**
+
+    ```bash
+    make build
+    ```
+
+1. **Run the `libvirt-provider`**
+   
+    The required libvirt-provider flags needs to be defined:
+
+    ```bash
+    go run provider/cmd/main.go \
+      --libvirt-provider-dir=<path-to-initialize-libvirt-provider> \
+      --supported-machine-classes=<path-to-machine-class-json>/machine-classes.json \
+      --network-interface-plugin-name=isolated \
+      --address=<local-path>/iri-machinebroker.sock
+    ```
+
+    Sample `machine-classes.json` can be found [here](../../config/development/machineclasses.json).
+
+## Interact with the `libvirt-provider`
+
+1. **Creating machine**
+
+    ```bash
+    irictl-machine --address=unix:<local-path-to-socket>/iri-machinebroker.sock create machine -f <path-to-machine-yaml>/iri-machine.yaml
+    ```
+
+    Sample `iri-machine.yaml`:
+
+    ```yaml
+        metadata:
```

Comment:
it looks little weird for me. Are you sure it is correct? I think metadata has to start directly under first `
Look on another code blocks, you don't have any extra whitespaces on beginning of line.

Reply:
ok. I will set the orientations properly

Reply:
done